apps/condo/domains/acquiring/schema/MultiPayment.js

```diff
@@ -150,20 +189,118 @@ const MultiPayment = new GQLListSchema('MultiPayment', {
+                const anotherCurrencyPayments = payments.filter(payment => payment.currencyCode !== currencyCode).map(payment => `"${payment.id}"`)
+                if (anotherCurrencyPayments.length) {
+                    return addValidationError(`${MULTIPAYMENT_MULTIPLE_CURRENCIES} [${anotherCurrencyPayments.join(', ')}]`)
+                }
+                const receiptsIds = new Set(payments.map(payment => payment.receipt))
+                if (receiptsIds.size !== payments.length) {
+                    return addValidationError(MULTIPAYMENT_NOT_UNIQUE_RECEIPTS)
+                }
+                const totalAmount = payments.reduce((acc, cur) => acc.plus(cur.amount), new Big(0))
+                if (!totalAmount.eq(resolvedData['amountWithoutExplicitFee'])) {
+                    return addValidationError(MULTIPAYMENT_TOTAL_AMOUNT_MISMATCH)
+                }
+                const contexts = await find('AcquiringIntegrationContext', {
+                    id_in: payments.map(payment => payment.context),
+                })
+                const integrations = new Set(contexts.map(context => context.integration))
+                if (integrations.size !== 1) {
+                    return addValidationError(MULTIPAYMENT_MULTIPLE_ACQUIRING_INTEGRATIONS)
+                }
+                const integrationId = contexts[0].integration
+                if (resolvedData['integration'] !== integrationId) {
+                    return addValidationError(MULTIPAYMENT_ACQUIRING_INTEGRATIONS_MISMATCH)
+                }
+                const integration = await getById('AcquiringIntegration', integrationId)
+                if (!integration.canGroupReceipts && payments.length > 1) {
+                    return addValidationError(MULTIPAYMENT_CANNOT_GROUP_RECEIPTS)
+                }
+            } else if (operation === 'update') {
+                const payments = await find('Payment', {
+                    multiPayment: { id: existingItem.id },
+                })
+                const oldStatus = existingItem.status
+                const newStatus = get(resolvedData, 'status', oldStatus)
+                if (!MULTIPAYMENT_TRANSITIONS[oldStatus].includes(newStatus)) {
+                    return addValidationError(`${MULTIPAYMENT_NOT_ALLOWED_TRANSITION} Cannot move from "${oldStatus}" status to "${newStatus}"`)
+                }
+                const newItem = {
+                    ...existingItem,
+                    ...resolvedData,
+                }
+                const requiredFields = MULTIPAYMENT_REQUIRED_FIELDS[newStatus]
+                let requiredMissing = false
+                for (const field of requiredFields) {
+                    if (!newItem.hasOwnProperty(field) || newItem[field] === null) {
+                        addValidationError(`${MULTIPAYMENT_MISSING_REQUIRED_FIELDS} Field ${field} was not provided`)
+                        requiredMissing = true
```

Comment:
if we don't need multiple errors here, please add `break`

Reply:
IMHO, it' better to give user all failed fields at the same time, rather than by 1....
So, I'm stacking errors and if flag is true returning